README.md

```diff
@@ -1,23 +1,70 @@
-     be in ruleset's account unless you [override the releaser](./templates/README.md#optional-configyml).
+Create a GitHub Actions workflow in your ruleset repository by creating a file, typically named `.github/workflows/publish.yaml`.
 
-   _Note: Authors of rulesets under the `bazelbuild` org should add the app to their personal fork of `bazelbuild/bazel-central-registry`._
+This repository provides a reusable workflow that contains all the boilerplate.
+See complete documentation in the [reusable workflow file](./.github/workflows/publish.yaml).
+
+1. Decide how your workflow will be invoked
+
+Use an `on` block, and provide at least the `tag_name` as an input.
+
+For example, if you have a release automation using GitHub Actions, you might call this publish workflow upon successful completion.
+A recommended setup for release automation, including generating attestations to prove the provenance of release artifacts, may be found at
+https://github.com/bazel-contrib/.github/blob/master/.github/workflows/
+
+As another example, you might use `workflow_dispatch` to manually run the publish workflow from the GitHub web UI or the CLI
+([documentation](https://docs.github.com/en/actions/managing-workflow-runs-and-deployments/managing-workflow-runs/manually-running-a-workflow))
+
+It's also useful to permit both of these, for example:
+
+```yaml
+on:
+  # Run the publish workflow after a successful release
+  # Will be triggered from the release.yaml workflow
+  workflow_call:
+    inputs:
+      tag_name:
+        required: true
+        type: string
+  # In case of problems, let release engineers retry by manually dispatching
+  # the workflow from the GitHub UI
+  workflow_dispatch:
+    inputs:
+      tag_name:
+        required: true
+        type: string
+```
+
+2. Reference the reusable workflow in your `publish` job (replacing the `[version]` placeholder).
+
+```yaml
+jobs:    
+  publish:
+    uses: bazel-contrib/publish-to-bcr/.github/workflows/publish.yaml@[version]
+    with:
+      tag_name: ${{ inputs.tag_name }}
+      # GitHub repository which is a fork of the upstream where the Pull Request will be opened.
+      registry_fork: aspect-build/bazel-central-registry
+    permissions:
+      contents: write  # allow creating a PR
+    secrets:
+      publish_token: ${{ secrets.PUBLISH_TOKEN }}
+```
 
-1. Include these [template files](./templates) in your ruleset repository.
-1. Cut a release. A pull request like [this](https://github.com/bazelbuild/bazel-central-registry/pull/1601) will be opened against the BCR.
+3. Create a Personal Access Token
+
+Save it as `PUBLISH_TOKEN` in your repository or org, under _Settings > Secrets and variables > Actions_.
```

Comment:
nit: add a note that this must be a classic PAT and not a fine-grained one, at least until https://github.com/github/roadmap/issues/600 is finished.